<launch>

<arg name="cam0_name" default="cam0"/>
<arg name="cam1_name" default="cam1"/>
<arg name="cam0_dev" value="/dev/video4"/>
<arg name="cam1_dev" value="/dev/video2"/>

<group ns="aruco_$(arg cam0_name)">
    <node name="$(arg cam0_name)" pkg="usb_cam" type="usb_cam_node" output="screen" >
      <param name="video_device" value="$(arg cam0_dev)"/>
      <param name="image_width" value="1920" />
      <param name="image_height" value="1080" />
      <param name="pixel_format" value="mjpeg" />
      <param name="color_format" value="yuv420p" />
      <param name="framerate" value="30" />
      <param name="camera_frame_id" value="$(arg cam0_name)" />
      <param name="io_method" value="mmap"/>
      <param name="camera_info_url" value="package://sisyph_slam/config/aceline_s200_1080p_30fps.yaml"/>
    </node>
</group>

<group ns="aruco_$(arg cam1_name)">
  <node name="$(arg cam1_name)" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="$(arg cam1_dev)"/>
    <param name="image_width" value="1280" />
    <param name="image_height" value="720" />
    <param name="pixel_format" value="mjpeg" />
    <param name="color_format" value="yuv422p" />
    <param name="framerate" value="30" />
    <param name="camera_frame_id" value="$(arg cam1_name)" />
    <param name="io_method" value="mmap"/>
    <param name="camera_info_url" value="package://sisyph_slam/config/hp_720p_30fps.yaml"/>
  </node>
</group>

</launch>
